vault/identity_store_oidc_provider.go

```diff
@@ -332,6 +419,19 @@ func (i *IdentityStore) pathOIDCDeleteScope(ctx context.Context, req *logical.Re
 	if err != nil {
 		return nil, err
 	}
+
+	targetScopeName := d.Get("name").(string)
+
+	providerNames, err := i.providersReferencingTargetScopeName(ctx, req, targetScopeName)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(providerNames) > 0 {
+		errorMessage := fmt.Sprintf("unable to delete scope %q because it is currently referenced by these providers: %s",
+			targetScopeName, strings.Join(providerNames, ", "))
+		return logical.ErrorResponse(errorMessage), logical.ErrInvalidRequest
```

Comment:
We should delete the scope (line 418) only after this provider reference check. I had the same bug in my PR: [link to change](https://github.com/hashicorp/vault/pull/12272/commits/f72b1d899b4c1d5f6d5fe6285401d18c1cb14b7c).

This will affect your UT as well. I will make a note there

Reply:
Fixed